src/Platforms/AbstractDB2Platform.php

```diff
@@ -0,0 +1,1016 @@
+<?php
+
+namespace Doctrine\DBAL\Platforms;
+
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\Exception;
+use Doctrine\DBAL\Schema\ColumnDiff;
+use Doctrine\DBAL\Schema\DB2SchemaManager;
+use Doctrine\DBAL\Schema\Identifier;
+use Doctrine\DBAL\Schema\Index;
+use Doctrine\DBAL\Schema\TableDiff;
+use Doctrine\DBAL\Types\Type;
+use Doctrine\DBAL\Types\Types;
+use Doctrine\Deprecations\Deprecation;
+
+use function array_merge;
+use function count;
+use function current;
+use function explode;
+use function func_get_arg;
+use function func_num_args;
+use function implode;
+use function sprintf;
+use function strpos;
+
+/**
+ * Provides the base implementation for the lowest versions of supported IBM DB2-like database platforms.
+ */
+abstract class AbstractDB2Platform extends AbstractPlatform
```

Comment:
What do we need this class for?

Reply:
The intention of this class is to provide the lowest base implementation for the supported IBM DB2 versions.

Reply:
Yeah, that's what the PHPDoc block says. But isn't this what `DB2Platform` already did? I don't understand why we need to create another base class.

Reply:
Yes, AFAIK, that's what the existing `DB2Platform` class does.
I was trying to mimic the approach used for MySQL, where the base implementation is provided through an abstract class:
https://github.com/doctrine/dbal/blob/bf27755bfe9f6a04f5fa7adb77ff30e204723ef3/src/Platforms/AbstractMySQLPlatform.php#L42
If you think that is not required in this case, I can revert this change and keep the original class.

Reply:
> I was trying to mimic the approach used for MySQL

We need this for MySQL because the platform splits into MySQL and MariaDB from there. We don't have this kind of situation for DB2, do we?

Reply:
The only class extending `AbstractMySQLPlatform` (at least, in the `src/` directory) is `MySQLPlatform`:
https://github.com/doctrine/dbal/blob/bf27755bfe9f6a04f5fa7adb77ff30e204723ef3/src/Platforms/MySQLPlatform.php#L9

`MariaDBPlatform` extends from `MySQLPlatform`:
https://github.com/doctrine/dbal/blob/bf27755bfe9f6a04f5fa7adb77ff30e204723ef3/src/Platforms/MariaDBPlatform.php#L11

Regarding the situation about IBM DB2, I think we currently don't have the same case, but I guess we should in the future (not a blocker for this PR). **IBM DB2** and **IBM DB2 for i** are different engines with different features (see https://github.com/SeidenGroup/doctrine-dbal-ibmi for example).

Reply:
> The only class extending `AbstractMySQLPlatform` (at least, in the `src/` directory) is `MySQLPlatform`:
> 
> https://github.com/doctrine/dbal/blob/bf27755bfe9f6a04f5fa7adb77ff30e204723ef3/src/Platforms/MySQLPlatform.php#L9
> 
> `MariaDBPlatform` extends from `MySQLPlatform`:

Yes, this was fixed in 4.0. Sorry, I should've been more precise. 🙂 

The fact that we branch into two separate platforms MySQL and MariaDB is the reason why there is a dedicated abstract platform class for MySQL but not for other platforms.

We don't need this kind of branching for DB2 at the moment, so we should not introduce it now.

> **IBM DB2** and **IBM DB2 for i** are different engines with different features

I am aware of that. But DBAL does not advertise support for DB2 for i and this PR does not introduce it. So my argument remains valid.

Reply:
The `AbstractDB2Platform` class was removed.

Reply:
Thanks! This makes the PR _way_ easier to review. 😅